extension-client-exec.md

```diff
@@ -0,0 +1,33 @@
+# `exec` extension to LSP
+
+The `exec` extension allows the server to send a command (e.g. `git blame`) to the client for execution, and the client responds with the stdout of the command.
+
+### `exec` Request
+
+_Request_:
+* method: 'xexec'
+* params: `ExecParams` defined as follows:
+```typescript
+interface ExecParams {
+    /**
+     * The name of the command to run
+     */
+    name: string;
+    /**
+     * The arguments to the command
+     */
+    arguments: string;
+}
+```
+
+_Response_:
+* result: `ExecResult`
+```typescript
+interface ExecResult {
+    /**
+     * The stdout of the process
+     */
+    stdout: string;
+}
+```
+* error: code and message set in case the process does not exit with code 0.
```

Comment:
It does not seem useful to treat an exec error as a protocol-level error. Let's reserve protocol-level errors for things like "do not know how to find the program" or "do not support execing". So if the process exits with a nonzero status code, its stdout and exit code should just be returned in the result data (same as if it exited with 0).